brokerapi/brokers/account_managers/sql_account_manager.go

```diff
@@ -163,6 +163,15 @@ func (sam *SqlAccountManager) pollOperationUntilDone(op *googlecloudsql.Operatio
 	return nil
 }
 
+func (b *SqlAccountManager) MergeCredentialsAndInstanceInfo(bindDetails map[string]string, instanceDetails map[string]string) map[string]string {
+	for key, val := range instanceDetails {
+		bindDetails[key] = val
```

Comment:
map is passed by reference so this will modify the callers object here. (IMO strange go behavior but I digress..)

maybe build up a totally fresh map here and return it?